OutSystems/outsystems-ui
# ROU-4364: Accessibility improvements identified on team audit for the Accordion

This PR is for improve the Accordion and AccordionItem accessibility

### What was happening
- The Accordion Content was focusable using the Tab key, but the Tab key is used to navigate between focusable elements
- When a clickable element is placed inside the Accordion Header, it is not clickable.
- The Accordion was structured as a TabList, which does not fit the best accessibility standards.
- Other improvements in terms of accessibility was needed.

### What was done
- Removed the tabindex=0 from the Accordion Content to prevent its focusability using the Tab key.
- Check if clickable elements are placed inside the Accordion Header:
    - If the Accordion is collapsed, we set they’re tab-index to -1 and aria-hidden to true.
    - If the Accordion is expanded, the tabindex is set to 0 and aria-hidden to false.
- Removed the tablist role from the Accordion and the tab an tabpanels roles from the AccordionItem.
- Other improvements: 
    - Changed the aria-disabled and aria-control from the AccordionItem wrapper to the Accordion Header button.
    - Added the region role to the AccordionItem Content element.
    - Removed the aria-expanded from the AccordionItem wrapper 


### Test Steps
1. Go to a page with an Accordion
2. Check if it meets the accessibility standards.


### Checklist
